src/screens/Departures/StopPlaceView.tsx

```diff
@@ -3,20 +3,24 @@ import Feedback from '@atb/components/feedback';
 import {useFavorites} from '@atb/favorites';
 import {UserFavoriteDepartures} from '@atb/favorites/types';
 import {DEFAULT_NUMBER_OF_DEPARTURES_PER_QUAY_TO_SHOW} from '@atb/screens/Departures/state/stop-place-state';
-import {SearchTime} from '@atb/screens/Departures/utils';
+import {
+  getLimitOfDeparturesPerLineByMode,
+  SearchTime,
+} from '@atb/screens/Departures/utils';
 import {StyleSheet} from '@atb/theme';
 import React, {useEffect, useMemo} from 'react';
 import {RefreshControl, SectionList, SectionListData, View} from 'react-native';
-import DateNavigation from './components/DateNavigator';
-import FavoriteToggle from './components/FavoriteToggle';
 import QuaySection from './components/QuaySection';
 import {useStopPlaceData} from './state/stop-place-state';
+import FavoriteToggle from '@atb/screens/Departures/components/FavoriteToggle';
+import DateSelection from '@atb/screens/Departures/components/DateSelection';
+import {PlaceScreenMode} from '@atb/screens/Departures/PlaceScreen';
 
 type StopPlaceViewProps = {
   stopPlace: Place;
   showTimeNavigation?: boolean;
   navigateToQuay: (quay: Quay) => void;
-  navigateToDetails: (
+  navigateToDetails?: (
```

Comment:
`navigateToDetails` is mandatory if `mode` is `'Departure', but should not be provided if `mode` is `Favourite`. It would be beneficial if this was reflected in the typing as well.